doc/OnlineDocs/contributed_packages/alternative_solutions.rst

```diff
@@ -88,6 +88,53 @@ Each ``Solution`` object contains information about the objective and variables,
    }
 
 
+Gap Usage Example
+-------------
+When we only want some of the solutions based off a tolerance away from optimal, this can be done using the ``abs_opt_gap`` parameter. This is shown in the following simple knapsack examples where the weights and values are the same.
+
+.. doctest::
+   :skipif: not glpk_available
+   
+   >>> import pyomo.environ as pyo
+   >>> import pyomo.contrib.alternative_solutions as aos
+
+   >>> values = [10,9,2,1,1]
+   >>> weights = [10,9,2,1,1]
+ 
+   >>> K = len(values)
+   >>> capacity = 12
+ 
+   >>> m = pyo.ConcreteModel()
+   >>> m.x = pyo.Var(range(K), within=pyo.Binary)
+   >>> m.o = pyo.Objective(expr=sum(values[i] * m.x[i] for i in range(K)), sense=pyo.maximize)
+   >>> m.c = pyo.Constraint(expr=sum(weights[i] * m.x[i] for i in range(K)) <= capacity) 
+ 
+   >>> solns = aos.enumerate_binary_solutions(m, num_solutions=10, solver="glpk", abs_opt_gap = 0.0)
+   >>> assert(len(solns) == 4)
+
+In this example, we only get the four ``Solution`` objects that have an ``objective_value`` of 12.
```

Comment:
(Not required), but it would be nice if this example showed the objective values for the 4 solutions (you say they are 12, but the reader has to take this on faith).  One solution might be to:
```diff
- >>> assert(len(solns) == 4)
+ >>> for soln in sorted(solns, key=lambda s: str(s.get_variable_name_values)):
+ >>>     print(soln)
+ {
+     "fixed_variables": [],
+     "objective": "o",
+     "objective_value": 12.0,
+     "solution": {
+         "x[0]": 0,
+         "x[1]": 1,
+         "x[2]": 1,
+         "x[3]": 0,
+         "x[4]": 1
+     }
+ }
+ {
+     "fixed_variables": [],
+     "objective": "o",
+     "objective_value": 12.0,
+     "solution": {
+         "x[0]": 0,
+         "x[1]": 1,
+         "x[2]": 1,
+         "x[3]": 1,
+         "x[4]": 0
+     }
+ }
+ {
+     "fixed_variables": [],
+     "objective": "o",
+     "objective_value": 12.0,
+     "solution": {
+         "x[0]": 1,
+         "x[1]": 0,
+         "x[2]": 1,
+         "x[3]": 0,
+         "x[4]": 0
+     }
+ }
+ {
+     "fixed_variables": [],
+     "objective": "o",
+     "objective_value": 12.0,
+     "solution": {
+         "x[0]": 1,
+         "x[1]": 0,
+         "x[2]": 0,
+         "x[3]": 1,
+         "x[4]": 1
+     }
+ }
```